### PR TITLE
Add findbugs to drone

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
This adds findbugs to drone:
- in https://nextcloud.kaminsky.me/index.php/s/fYZa7NeBsnmFZQD will be a result for each PR/master.
- for master there will be a findbugs.html, which shows the current warning counts (once it is merged)

In comparison to lint, there is no failure if count is increasing. 
Currently this is just for awareness.

To increase awareness, drone is pushing a comment with lint and findbugs stats.

Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### CODE REVIEW

**Comment by tobiasKaminsky** [comment]
test

File: scripts/analysis/analysis-wrapper.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        6="master-"$(date +%F)
+    fi
+    echo "New lint results at https://nextcloud.kaminsky.me/index.php/s/tXwtChzyqMj6I8v ->" $6.html
+    curl -u $4:$5 -X PUT https://nextcloud.kaminsky.me/remote.php/webdav/droneLogs/$6.html --upload-file build/reports/lint/lint.html
+    
+    echo "New findbugs results at https://nextcloud.kaminsky.me/index.php/s/fYZa7NeBsnmFZQD ->" $6.html
+    curl -u $4:$5 -X PUT https://nextcloud.kaminsky.me/remote.php/webdav/findbugs/$6.html --upload-file build/reports/findbugs/findbugs.html
+    
+    # add comment with results
+    lintResult="<h1>Lint</h1><table width='500' cellpadding='5' cellspacing='2'><tr class='tablerow0'><td>"$(tail -n1 scripts/analysis/lint-results.txt | cut -f2 -d':' |cut -f1 -d'<')"</td></tr></table>"
```
[the remark this answers is not in the document]
Comment (tobiasKaminsky) [reply scripts/analysis/analysis-wrapper.sh]: Please ignore this. It is ok to have nested quotes.

File: scripts/analysis/analysis-wrapper.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    findbugsResult=$(sed -n "/<h1>Summary<\/h1>/,/<h1>Warnings<\/h1>/p" build/reports/findbugs/findbugs.html |head -n-1 | sed s'/<\/a>//'g | sed s'/<a.*>//'g | sed s'/Summary/FindBugs/' | tr "\"" "\'" | tr -d "\n")
```
[the remark this answers is not in the document]
Comment (tobiasKaminsky) [reply scripts/analysis/analysis-wrapper.sh]: No idea what codacy is complaining about, but works for me/drone ;-)

[the remark this answers is not in the document]
Comment (AndyScherzinger) [reply scripts/analysis/analysis-wrapper.sh]: put on ignore list

File: scripts/analysis/analysis-wrapper.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (AndyScherzinger) [reply scripts/analysis/analysis-wrapper.sh]: put on ignore list

**Comment by AndyScherzinger** [comment]
:+1: fine by me, codacy warning will be gone with next build :)

**Comment by tobiasKaminsky** [comment]
Some more improvements to come, so please do not merge right now.

File: .drone.yml [hunk 3]
```diff
@@ -1,17 +1,17 @@
 pipeline:
   compile:
-    image: nextcloudci/android:android-35
+    image: nextcloudci/android:android-36
```
Comment (AndyScherzinger) [inline .drone.yml]: Nice new docker release :)

**Comment by AndyScherzinger** [comment]
:+1: nice addition, not sure why lint score is down by 3 due to this PR though ;)

**Comment by tobiasKaminsky** [comment]
I had to create a new docker-ci version, to add a program, but there the emulator fails.
So for now, we are using the old image (35) for testing, but the newer one (37) for analytics.

--> Ready for review :+1:

**Comment by mario** [comment]
👍 
Please add this to Talk too.

**Comment by tobiasKaminsky** [comment]
Will do it, but it may take a bit.

**Pull request merged**